docs/api/python/getting-started.md

```diff
@@ -207,324 +369,882 @@ if __name__ == "__main__":
+
+    # Get current gas price in Wei
+    gas_price = zk_web3.zksync.gas_price
+
+    # Create deployment contract transaction
+    create_contract = TxCreateContract(
+        web3=zk_web3,
+        chain_id=chain_id,
+        nonce=nonce,
+        from_=account.address,
+        gas_limit=0,  # UNKNOWN AT THIS STATE,
+        gas_price=gas_price,
+        bytecode=incrementer_contract.bytecode,
+        call_data=encoded_constructor,
+        salt=random_salt
+    )
+
+    # ZkSync transaction gas estimation
+    estimate_gas = zk_web3.zksync.eth_estimate_gas(create_contract.tx)
+    print(f"Fee for transaction is: {Web3.from_wei(estimate_gas * gas_price, 'ether')} ETH")
+
+    # Convert transaction to EIP-712 format
+    tx_712 = create_contract.tx712(estimate_gas)
+
+    # Sign message
+    signed_message = signer.sign_typed_data(tx_712.to_eip712_struct())
+
+    # Encode signed message
+    msg = tx_712.encode(signed_message)
+
+    # Deploy contract
+    tx_hash = zk_web3.zksync.send_raw_transaction(msg)
+
+    # Wait for deployment contract transaction to be included in a block
+    tx_receipt = zk_web3.zksync.wait_for_transaction_receipt(
+        tx_hash, timeout=240, poll_latency=0.5
+    )
+
+    print(f"Tx status: {tx_receipt['status']}")
+    contract_address = tx_receipt["contractAddress"]
+    print(f"contract address: {contract_address}")
+
+    # Check does precompute address match with deployed address
+    if precomputed_address.lower() != contract_address.lower():
+        raise RuntimeError("Precomputed contract address does now match with deployed contract address")
+
+    return contract_address
+
+
+def execute(
+        zk_web3: Web3, account: LocalAccount, compiled_contract: Path, contract_address: HexAddress
+):
+    """Interact with deployed smart contract on zkSync network
+
+    :param zk_web3:
+        Instance of ZkSyncBuilder that interacts with zkSync network
+
+    :param account:
+        From which account the deployment contract tx will be made
+
+    :param compiled_contract:
+        Compiled contract source.
+
+    :param contract_address:
+        Contract address on zkSync network
+    """
+    # Get contract ABI and bytecode information
+    incrementer_contract = ContractEncoder.from_json(zk_web3, compiled_contract)[0]
+
+    # Execute Get method on smart contract
+    value = incrementer_contract.contract.functions.get().call(
+        {
+            "from": account.address,
+            "to": contract_address
+        })
+    print(f"Value: {value}")
+
+    gas_price = zk_web3.zksync.gas_price
+
+    # Get nonce of ETH address on zkSync network
+    nonce = zk_web3.zksync.get_transaction_count(account.address, EthBlockParams.LATEST.value)
+
+    # Execute increment method on smart contract
+    tx = incrementer_contract.contract.functions.increment().build_transaction({
+        "nonce": nonce,
+        "from": account.address,
+        "maxPriorityFeePerGas": 1000000,
```

Comment:
can you explain this a little bit more? if maxFee is set to be the current gas price - then this doesn't have much effect, right?

Reply:
Every transaction on L2 has` maxFeePerGas: gasPrice`. It's important that  gas price must be higher than: `base fee + maxPriorityFeePerGas`. Gas price on L2 is 250.000.000 which is high enough and all unused gas is return to the user.